### PR TITLE
Part one: add wait messages in some of our pages

### PR DESCRIPTION
**Pull request by AlexandraMoga**
I've started adding wait messages in the following pages:

- `pages/desktop/base.py`
- `pages/desktop/login.py`
- `pages/desktop/reviews.py`
- `pages/desktop/search.py`

I didn't want to have a huge patch, so the other pages will be handled in other patches.

I've also tweaked the waits in two functions:  `pages/desktop/base.py - logged_in` and `pages/desktop/reviews.py - click_reply_to_review`

### CODE REVIEW

File: pages/desktop/base.py [hunk 1]
```diff
@@ -48,18 +53,15 @@ def logged_in(self):
...
-                )
-            )
-            return self.is_element_displayed(*self.header._user_locator)
+        count = 0
+        while count < 5:
+            try:
+                self.is_element_displayed(*self.header._user_locator)
+                break
+            except StaleElementReferenceException as exception:
+                print(f'{exception}: Trying to find the element again')
```
Comment (jrbenny35) [inline pages/desktop/base.py:62]: Should this be "Try to find the element again"?

File: pages/desktop/base.py [hunk 2]
```diff
@@ -247,19 +259,29 @@ def more_menu(self, item=None):
 
     def click_developer_hub(self):
         self.find_element(*self._devhub_locator).click()
-        self.wait.until(EC.number_of_windows_to_be(2))
+        self.wait.until(
+            EC.number_of_windows_to_be(2),
+            message=f'Number of windows was {len(self.selenium.window_handles)}',
```
Comment (jrbenny35) [inline pages/desktop/base.py:264]: could we add an "expected 2" at the end of this?
```suggestion
            message=f'Number of windows was {len(self.selenium.window_handles)}, expected 2',
```

File: pages/desktop/base.py [hunk 3]
```diff
@@ -247,19 +259,29 @@ def more_menu(self, item=None):
...
+            EC.visibility_of_element_located((By.CLASS_NAME, 'DevHub-Navigation-Logo')),
+            message=f'DevHub homepage not loaded; page was {self.selenium.current_url}',
         )
 
     def click_extension_workshop(self):
         self.find_element(*self._extension_workshop_locator).click()
-        self.wait.until(EC.number_of_windows_to_be(2))
+        self.wait.until(
+            EC.number_of_windows_to_be(2),
+            message=f'Number of windows was {len(self.selenium.window_handles)}',
```
Comment (jrbenny35) [inline pages/desktop/base.py:277]: could we add an "expected 2" at the end of this?
```suggestion
            message=f'Number of windows was {len(self.selenium.window_handles)}, expected 2',
```

File: pages/desktop/base.py [hunk 4]
```diff
@@ -22,11 +22,16 @@ def __init__(self, selenium, base_url, locale='en-US', **kwargs):
...
+        self.wait.until(
+            lambda _: self.find_element(*self._amo_header).is_displayed(),
+            message='AMO header was not loaded',
+        )
         return self
 
     def wait_for_title_update(self, term):
-        self.wait.until(EC.title_contains(term))
+        self.wait.until(
+            EC.title_contains(term), message=f'Page title was {self.selenium.title}'
```
Comment (jrbenny35) [inline pages/desktop/base.py:33]:
```suggestion
            EC.title_contains(term), message=f'Page title was {self.selenium.title}, expected{term}.'
```

Thoughts?

**Review by jrbenny35 — CHANGES_REQUESTED** [review]
Love these, will help debugging for sure!

**Comment by AlexandraMoga** [comment]
Change requests solved. I'm merging this PR because I need those debug messages up and running.

**Pull request merged**